### ISSUE TITLE
Can I use OMNeT++ 6.0  in COSSIM_framework?

### ISSUE THREAD

**Issue by 2024slby** [H2020-COSSIM/COSSIM_framework#4]
Hi,I think your work is very meaningful and helpful to me!
Now I want to simulate tsn which OMNeT++ 6.0 support. Is it easy to achieve this? And what should I do to achieve this?Thanks.

**Comment by kingmouf** [comment]
Hi! Thank you for the interest in the project! It sure is possible to do that and I propose that you look at the overall documentation provided. Also you can contact directly the main maintainer of the project @ntampouratzis (you can also find his email in the repository).

**Comment by ntampouratzis** [comment]
Hello! Thank you for your interest for the COSSIM project. We have tried in the past to integrate COSSIM with OMNeT++ 6.0 but unfortunately, there are a lot of changes in the major revision  OMNeT++ 6.x. The main elimination is that there is not AODV Routing implementation in OMNeT++ 6.0, as a result, our micro-router functionality which converts Ethernet packets to different network protocols is not working. So, currently we have managed to support only L2 routing (Ethernet Routing) in OMNeT++ 6.0.

**Comment by 2024slby** [comment]
Thanks for your replies! It's truly exciting to see that you've already tried to integrate COSSIM with OMNeT++ 6.0.
I have read most of documents.So now the problem is that the data packets sent by cgem5 cannot be converted into other protocols except Ethernet. Therefore, currently, it is not possible to simulate TSN, right?

**Comment by ntampouratzis** [comment]
Yes, you are right we cannot convert Ethernet packets into other protocols in OMNeT++ 6.0. Specifically, cgem5 creates standard real Ethernet Packets from the simulated NIC. Currently (in OMNeT++ 6.0) we decapsulate the Ethernet packets to take the Ethernet header (source, destination, etc.) to route the packets only through Ethernet Switch.

**Comment by 2024slby** [comment]
OK,I see,thanks for your patience. Since I just want to simulate an easy TSN network,and TSN is based on Ethernet, I think maybe I can do something to achieve this(as long as INET 4.4 or above version can be used).May I have the honor to study the code that integrates with OMNET++ 6.0?

**Comment by ntampouratzis** [comment]
I have just uploaded the OMNET++ Workspace with INET 4.4.1 (https://github.com/ntampouratzis/COSSIM-OMNETv6.git).
First of all, I suggest to download our VM (https://github.com/H2020-COSSIM/COSSIM_framework) which contains all appropriate COSSIM subtools with the dependencies, OS images, and kernels. Subsequently, you need to install OMNET v6.0 on it (remove the OMNET v5.0) and use the above Workspace. In this Workspace we have set up a simple example with 2 gem5 nodes (OMNET_WORKSPACE/test/simulations/ARPTest.ned). The main file that receives the gem5 packet and sends it through OMNET++ is the Txc.cc (OMNET_WORKSPACE/HLANode/src/Txc.cc). To be noticed that this file was written by an undergraduate student who tried to integrate the COSSIM with OMNET v6.0 before 2 years.

**Comment by 2024slby** [comment]
Thank you very much!
